node/collation-generation/src/lib.rs

```diff
@@ -141,11 +140,10 @@ impl CollationGenerationSubsystem {
 			}) => {
 				if self.config.is_some() {
 					log::warn!(target: LOG_TARGET, "double initialization");
-					true
 				} else {
 					self.config = Some(Arc::new(config));
-					false
 				}
+				false
```

Comment:
I think this could be a legitimate reason to stop everything. However, I would make the warning above an error than.